test/extended/baremetal/high_availability.go

```diff
@@ -0,0 +1,135 @@
+package baremetal
+
+import (
+	"context"
+	"strings"
+	"time"
+
+	e2e "k8s.io/kubernetes/test/e2e/framework"
+
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	g "github.com/onsi/ginkgo"
+	o "github.com/onsi/gomega"
+	exutil "github.com/openshift/origin/test/extended/util"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	deleteInterval     = 1 * time.Second
+	deleteWaitTimeout  = 3 * time.Minute
+	restartInterval    = 10 * time.Second
+	restartWaitTimeout = 5 * time.Minute
+	baremetalNamespace = "openshift-machine-api"
+)
+
+var _ = g.Describe("[sig-installer][Feature:baremetal][Serial] Baremetal platform should ensure", func() {
+	defer g.GinkgoRecover()
+
+	oc := exutil.NewCLI("baremetal")
+
+	g.It("cluster baremetal operator and metal3 deployment return back healthy after they are deleted", func() {
+		skipIfNotBaremetal(oc)
+
+		c, err := e2e.LoadClientset()
+		o.Expect(err).ToNot(o.HaveOccurred())
+
+		g.By("delete cluster baremetal operator")
+		err = c.AppsV1().Deployments(baremetalNamespace).Delete(context.Background(), "cluster-baremetal-operator", v1.DeleteOptions{})
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		g.By("wait until cluster baremetal operator is deleted")
+		err = wait.PollImmediate(deleteInterval, deleteWaitTimeout, func() (bool, error) {
+			_, err := c.AppsV1().Deployments(baremetalNamespace).Get(context.Background(), "cluster-baremetal-operator", v1.GetOptions{})
+			if err != nil {
+				if strings.Contains(err.Error(), `deployments.apps "cluster-baremetal-operator" not found`) {
```

Comment:
I think `if errors.IsNotFound(err) {` could be used for that

Reply:
Thanks. Done.